main.go

```diff
@@ -740,15 +743,19 @@ func (d *dump) dump(fc chan<- sumFileJob) error {
 	d.Path = file
 	d.ExitCode = 0
 
-	var mode os.FileMode = 0600
-	if d.Options.Format == 'd' {
-		// The hardening of permissions only apply to the top level
-		// directory, this won't make the contents executable
-		mode = 0700
-	}
+	var mode os.FileMode = os.FileMode(d.Mode)
+	if d.Mode > 0 {
+		if d.Options.Format == 'd' {
+			// The hardening of permissions only apply to the top level
+			// directory, this won't make the contents executable
+			// we keep this for compatibility reason, but since file mode is under
+			// user control, we should probably remove this
+			mode = 0700
```

Comment:
The directory format should be handled as best as possible. I'd imagine a user would expect that if mode is set to 0640, the directory would be chmod  to 0750 and its contents to 0640.

On the file, it's ease, just walk the contents of the directory and os.Chmod to the given mode.

Deducing the mode of the directory is trickier. What do you think of "grant x when r or w is present in the mode for each category of users"?

I mean:
```
if (mode & 0o400 > 0) || (mode & 0o200 > 0) {
	mode = mode | 0o100
}

/// and os on to add 0o010 and 0o001
```

Reply:
done this way, i think it's a first good approach... but may be we should add a specific settings to configure the permission on the root directory when using the `directory` format.

Reply:
The update of the mode should be done also for the "group" and "other" bits, in a similar way. Apart from that, LGTM.

Reply:
IMO, just updating the permission for the directory like this (for the group and other as well) is enough to have something that work for the vast majority of use cases. So no need to add an extra configuration parameter for the directory format.